scrapy/extensions/httpcache.py

```diff
@@ -303,41 +305,41 @@ def retrieve_response(self, spider, request):
         response = respcls(url=url, headers=headers, status=status, body=body)
         return response
 
-    def store_response(self, spider, request, response):
+    def store_response(self, spider: Spider, request: Request, response):
         """Store the given response in the cache."""
         rpath = self._get_request_path(spider, request)
-        if not os.path.exists(rpath):
-            os.makedirs(rpath)
+        if not rpath.exists():
+            rpath.mkdir(parents=True)
         metadata = {
             'url': request.url,
             'method': request.method,
             'status': response.status,
             'response_url': response.url,
             'timestamp': time(),
         }
-        with self._open(os.path.join(rpath, 'meta'), 'wb') as f:
+        with self._open(rpath / 'meta', 'wb') as f:
             f.write(to_bytes(repr(metadata)))
-        with self._open(os.path.join(rpath, 'pickled_meta'), 'wb') as f:
+        with self._open(rpath / 'pickled_meta', 'wb') as f:
             pickle.dump(metadata, f, protocol=4)
-        with self._open(os.path.join(rpath, 'response_headers'), 'wb') as f:
+        with self._open(rpath / 'response_headers', 'wb') as f:
             f.write(headers_dict_to_raw(response.headers))
-        with self._open(os.path.join(rpath, 'response_body'), 'wb') as f:
+        with self._open(rpath / 'response_body', 'wb') as f:
             f.write(response.body)
-        with self._open(os.path.join(rpath, 'request_headers'), 'wb') as f:
+        with self._open(rpath / 'request_headers', 'wb') as f:
             f.write(headers_dict_to_raw(request.headers))
-        with self._open(os.path.join(rpath, 'request_body'), 'wb') as f:
+        with self._open(rpath / 'request_body', 'wb') as f:
             f.write(request.body)
 
-    def _get_request_path(self, spider, request):
+    def _get_request_path(self, spider: Spider, request: Request) -> Path:
         key = self._fingerprinter.fingerprint(request).hex()
-        return os.path.join(self.cachedir, spider.name, key[0:2], key)
+        return Path(self.cachedir, spider.name, key[0:2], key)
```

Comment:
That's a risky change; even if a method is declared as private, it's still overridden in the wild, e.g. https://github.com/scrapy-plugins/scrapy-splash/blob/f5273b3a0ef15a9f6809d305a1dbe8f3efd12c5f/scrapy_splash/cache.py#L17

Reply:
I guess the question is, is it private as in it's not supposed to be called by other types or is it private as in it's part of Scrapy and if you touch it you're accepting the risk of it unexpectedly changing? I can put a `str()` around it.